drivers/sensor/bmi08x/bmi08x_accel_trigger.c

```diff
@@ -0,0 +1,168 @@
+/* Bosch BMI08X inertial measurement unit driver, trigger implementation
+ *
+ * Copyright (c) 2022 Meta Platforms, Inc. and its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/kernel.h>
+
+#define DT_DRV_COMPAT bosch_bmi08x_accel
+#include "bmi08x.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(BMI08X_ACCEL, CONFIG_SENSOR_LOG_LEVEL);
+
+static void bmi08x_handle_drdy_acc(const struct device *dev)
+{
+	struct bmi08x_accel_data *bmi08x = dev->data;
+	struct sensor_trigger drdy_trigger = {
+		.type = SENSOR_TRIG_DATA_READY,
+	};
```

Comment:
Store a reference to the user-supplied trigger in `bmi08x_trigger_set_acc` and pass it back when invoking the handler. This will allow the application to use CONTAINER_OF with the trigger